rasa_sdk/tracing/trace_provider.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional
+from rasa_sdk.utils import Singleton
+from opentelemetry.trace import Tracer
+
+
+class TraceProvider(metaclass=Singleton):
+    """Represents a provider for tracer."""
+
+    tracer: Optional[Tracer] = None
```

Comment:
Hmmm...since this will always refer to the same data, what's the point of the dictionary? 🤔

Reply:
Or do we wanna have multiple tracers based on the instrumented class module?

Reply:
> have multiple tracers based on the instrumented class module?

Yup in case of that. We should add docstrings to document this too.

Reply:
This will make things a bit complicated for the users having to know which tracer to use.

Having multiple tracer options based on the instrumented class module won't really change much. Only these tags are dependent on the tracer used. The positioning of the span doesn't change.
<img width="273" alt="Screenshot 2024-02-13 at 17 09 11" src="https://github.com/RasaHQ/rasa-sdk/assets/23738768/b62e4f40-b57c-4290-8b04-d01edab41ba9">

Also instrumented methods in the code may or may not be called by the custom action. That's why I chose the ActionExecutor since it's a common factor.

Reply:
Ok, in that case we need to make it explicit in the name and docstring that this trace register is only for the ActionExecutor's tracer e.g. `ActionExecutorTracerRegister`.